lib/Pipes/PipelineManager.cpp

```diff
@@ -0,0 +1,294 @@
+#include <list>
+#include <memory>
+#include <string>
+
+#include "llvm/ADT/ArrayRef.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/Support/Error.h"
+#include "llvm/Support/TargetSelect.h"
+#include "llvm/Support/raw_os_ostream.h"
+
+#include "revng/Model/LoadModelPass.h"
+#include "revng/Pipeline/Context.h"
+#include "revng/Pipeline/Loader.h"
+#include "revng/Pipeline/Registry.h"
+#include "revng/Pipeline/Runner.h"
+#include "revng/Pipeline/Target.h"
+#include "revng/Pipes/PipelineManager.h"
+#include "revng/Support/ResourceFinder.h"
+
+using namespace pipeline;
+using namespace std;
+using namespace llvm;
+
+const char LLVMContextWrapper::ID = '0';
+
+class LoadModelPipePass {
+private:
+  ModelWrapper *Wrapper;
+
+public:
+  static constexpr auto Name = "Load Model";
+  std::vector<ContractGroup> getContract() const { return {}; }
```

Comment:
`PureLLVMPipe`?

Reply:
same issue as other one above.

Reply:
this one is magical and must be scheduled in front of every llvm pipeline so that they can access the model